### PR TITLE
WT-7038 Update headers to be C++ compatible

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

**Comment by luke-pearson** [comment]
Since the original change was merged, two fixes have been added:

1. The public API struct nesting format has been restored, in order to support it an ifdef was added.
2. C++ standard 11 is now specified in the Makefile for the test_harness and a cppstyle c header was added to the top of test_harness.h (this fixes compilation on gcc4.8).

I've added the comments to reference the changes specifically.

File: src/include/packing_inline.h [hunk 1]
```diff
@@ -113,7 +113,11 @@ __pack_name_next(WT_PACK_NAME *pn, WT_CONFIG_ITEM *name)
         WT_CLEAR(*name);
         name->str = pn->buf;
         name->len = strlen(pn->buf);
+#ifdef __cplusplus
```
Comment (luke-pearson) [inline src/include/packing_inline.h:116]: Required to build on C++ and C.

File: test/cppsuite/Makefile.am [hunk 2]
```diff
@@ -1,4 +1,4 @@
-AM_CPPFLAGS = -I$(top_builddir) -I$(top_srcdir)/src/include \
+AM_CPPFLAGS = -std='c++11' -I$(top_builddir) -I$(top_srcdir)/src/include \
```
Comment (luke-pearson) [inline test/cppsuite/Makefile.am:1]: Needed by gcc-4.8

File: test/cppsuite/test_harness/test_harness.h [hunk 3]
```diff
@@ -2,8 +2,12 @@
 #ifndef TEST_HARNESS_H
 #define TEST_HARNESS_H
 
+/* Required to build using older versions of g++. */
+#include <cinttypes>
```
Comment (luke-pearson) [inline test/cppsuite/test_harness/test_harness.h:6]: Needed by gcc-4.8

**Review by ddanderson — APPROVED** [review]
LGTM.

File: src/include/packing_inline.h (hunk 1 above)
Comment (agorrod) [reply src/include/packing_inline.h:116]: I think this deserves a comment explaining why it's here. Something like "C++ treats access to nested data structures differently than C, so we need to refer to them differently depending on the compiler."

Comment (luke-pearson) [reply src/include/packing_inline.h:116]: Good point, that should've had a comment.

Comment (luke-pearson) [reply src/include/packing_inline.h:120]: 7936a12

**Review by agorrod — APPROVED** [review]
lgtm

**Pull request merged**